app/src/main/java/org/simple/clinic/encounter/Encounter.kt

```diff
@@ -33,6 +36,24 @@ data class Encounter(
   interface RoomDao {
 
     @Insert(onConflict = REPLACE)
-    fun save(encounters: List<Encounter>)
+    fun save(encounters: List<Encounter>): Completable
+
+    @Query("UPDATE Encounter SET syncStatus = :newStatus WHERE syncStatus = :oldStatus")
+    fun updateSyncStatus(oldStatus: SyncStatus, newStatus: SyncStatus): Completable
+
+    @Query("UPDATE Encounter SET syncStatus = :newStatus WHERE uuid IN (:uuids)")
+    fun updateSyncStatus(uuids: List<UUID>, newStatus: SyncStatus): Completable
+
+    @Query("SELECT * FROM Encounter WHERE syncStatus = :syncStatus")
+    fun recordsWithSyncStatus(syncStatus: SyncStatus): Observable<List<Encounter>>
+
+    @Query("SELECT COUNT(uuid) FROM Encounter")
+    fun recordCount(): Observable<Int>
+
+    @Query("SELECT COUNT(uuid) FROM Encounter WHERE syncStatus = :syncStatus")
+    fun recordCount(syncStatus: SyncStatus): Observable<Int>
+
+    @Query("SELECT * FROM Encounter WHERE uuid = :uuid")
+    fun getOne(uuid: UUID): Encounter?
```

Comment:
Should this return type be a `Maybe<Encounter>` or `Single<Encounter>` (depending upon what is supported by room)?

Reply:
The usage of this method does not require this to be reactive.
Also, we only expect one value here since the `WHERE` clause is on `uuid` which is a `PRIMARY KEY`.
I don't think it is necessary to use `Maybe` or `Single` here.